src/Button.tsx

```diff
@@ -1,29 +1,42 @@
-import React, { useContext, ReactNode } from 'react'
 import classNames from 'classnames'
-import warn from './utils/warning'
+import React, { ReactNode, useContext } from 'react'
 import { ThemeContext } from './context/ThemeContext'
+import warn from './utils/warning'
+
+type IconType =
+  | string
+  | React.FunctionComponent<{ className: string; 'aria-hidden': boolean }>
+  | React.ComponentClass<{ className: string; 'aria-hidden': boolean }>
 
 interface Props {
   children?: React.ReactNode
-  tag?: string
   disabled?: boolean
   size?: 'larger' | 'large' | 'regular' | 'small' | 'pagination'
-  // NEEDS REVIEW - could it be more specific?
-  icon?: any
-  iconLeft?: any
-  iconRight?: any
+  icon?: IconType
+  iconLeft?: IconType
+  iconRight?: IconType
   layout?: 'outline' | 'link' | 'primary' | '__dropdownItem'
-  type?: 'button' | 'submit' | 'reset'
   block?: boolean
-  className?: string
-  onClick?: React.MouseEvent
 }
 
+export interface ButtonAsButtonProps extends Props, React.ButtonHTMLAttributes<HTMLButtonElement> {
+  tag?: 'button'
+  type?: 'button' | 'submit' | 'reset'
+}
+
+export interface ButtonAsAnchorProps extends Props, React.AnchorHTMLAttributes<HTMLAnchorElement> {
+  tag: 'a'
+}
+
+export type ButtonProps = ButtonAsButtonProps | ButtonAsAnchorProps
```

Comment:
What about `react-router` Links?

Current usage: https://github.com/estevanmaito/windmill-dashboard-react/blob/master/src/pages/Buttons.js#L33

`react-router` supports a `component` prop, but it expects an anchor tag, and in our case we render a `button` as default. It could be used like `<Link to="/" component={Button} />` if there was a way.

Reply:
Maybe add additional exports with pre-bound tag prop?

```ts
import React from 'react'
import { ButtonAnchor as Button } from '@windmill/react-ui'

const CustomLink: React.FC = () => <Link to="/" component={Button} />

```

Would make things a little more convenient. Otherwise the developer can wrap it themselves.

Reply:
resolved in 1df0f17